qiskit_experiments/data_processing/nodes.py

```diff
@@ -34,65 +37,67 @@ def __init__(self, axis: int, validate: bool = True):
         super().__init__(validate)
         self._axis = axis
 
-    def _format_data(self, datum: Any, error: Optional[Any] = None):
-        """Format the data into numpy arrays."""
-        datum = np.asarray(datum, dtype=float)
+    def _format_data(self, data: np.ndarray) -> np.ndarray:
+        """Format the data into numpy arrays.
+
+        Args:
+            data: A full data array to format.
 
+        Returns:
+            The data that has been validated and formatted.
+
+        Raises:
+            DataProcessorError: When the specified axis does not exist in given array.
+        """
         if self._validate:
-            if len(datum.shape) <= self._axis:
+            if len(data.shape) <= self._axis:
                 raise DataProcessorError(
-                    f"Cannot average the {len(datum.shape)} dimensional "
+                    f"Cannot average the {len(data.shape)} dimensional "
                     f"array along axis {self._axis}."
                 )
 
-        if error is not None:
-            error = np.asarray(error, dtype=float)
+        return data
 
-        return datum, error
+    def _process(self, data: np.ndarray) -> np.ndarray:
+        r"""Average the data.
 
-    def _process(
-        self, datum: np.array, error: Optional[np.array] = None
-    ) -> Tuple[np.array, np.array]:
-        """Average the data.
-
-         Args:
-             datum: an array of data.
+        Args:
+            data: A full data array to format.
 
-         Returns:
-             Two arrays with one less dimension than the given datum and error. The error
-             is the standard error of the mean, i.e. the standard deviation of the datum
-             divided by :math:`sqrt{N}` where :math:`N` is the number of data points.
+        Notes:
+            Error is computed by the standard error of the mean,
+            i.e. the standard deviation of the datum divided by :math:`\sqrt{N}`
+            where :math:`N` is the number of data points.
+            Standard error computed by the previous node will be discarded.
```

Comment:
```suggestion
            The error is computed by the standard error of the mean,
            i.e. the standard deviation of the datum divided by :math:`\sqrt{N}`
            where :math:`N` is the number of data points.
            Standard errors computed by the previous node are discarded.
```

Reply:
Does it always have to be the case that errors from the previous node are discarded? Propagating errors through an averaging should be easy.

Reply:
Yes, but this violate unittest. I understand you designed the node to compute variance of nominal values, not the error propagation through sum.

```python
vals = unp.uarray([1,2,3,4,5,6], [0.1,0.2,0.3,0.4,0.5,0.6])
>>> np.sum(vals/6)
3.5+/-0.15898986690282427

>>> np.std(unp.nominal_values(vals)) / np.sqrt(6)
0.6972166887783964
```
These are the different quantities.

Reply:
Ok, I think we should change the node to use the uncertainties package. Looking at https://en.wikipedia.org/wiki/Propagation_of_uncertainty the current formula only holds if all variances are equal (see the end of the `Linear combinations` section). What uncertainty is returned from the sum if there is no standard deviation given in the `unp.array`? I.e. what does this do?
```
vals = unp.uarray([1,2,3,4,5,6])
>>> np.sum(vals/6)
3.5+/-??
```

Reply:
This becomes zero or nan depending on what you set to stdev (note that you cannot initialize uarray without stdev). This just computes quadratic sum of stdevs.

Reply:
Done in https://github.com/Qiskit/qiskit-experiments/pull/481/commits/19ff8f1f2a7c246931a6a789d9d327fd85e1e7c7